File: src/main/java/dev/salmon/keystrokes/hud/GuiKeySpace.java
package dev.salmon.keystrokes.hud;

import dev.salmon.keystrokes.hud.api.KeyRenderer;
import net.minecraft.client.settings.KeyBinding;

public class GuiKeySpace extends GuiKey {
    public GuiKeySpace(float x, float y, float width, float height, KeyBinding keyBinding) {
        super(x, y, width, height, keyBinding);
    }

    @Override
    public void render(float baseX, float baseY, float scale) {
        float x = baseX + relX * scale;
        float y = baseY + relY * scale;
        float w = width * scale;
        float h = height * scale;

        KeyRenderer.drawBackground(x, y, w, h, style, isPressed, percentFaded, scale);

        int color = style.textColor(isPressed, percentFaded);
        boolean shadow = isPressed ? style.isShadowActive() : style.isShadowIdle();
        KeyRenderer.drawSpaceGlyph(x, y, w, h, color, shadow);
    }

    @Override
    protected String getKeyName() {
        return ""; // not used for space rendering
    }
}

File: src/main/kotlin/org/polyfrost/keystrokes/hud/GuiKeySpace.kt
package org.polyfrost.keystrokes.hud

import net.minecraft.client.settings.KeyBinding
import org.polyfrost.keystrokes.hud.api.KeyRenderer.drawBackground
import org.polyfrost.keystrokes.hud.api.KeyRenderer.drawSpaceGlyph

class GuiKeySpace(x: Float, y: Float, width: Float, height: Float, keyBinding: KeyBinding) : GuiKey(x, y, width, height, keyBinding) {
    override fun render(baseX: Float, baseY: Float, scale: Float) {
        val x = baseX + relX * scale
        val y = baseY + relY * scale
        val w = width * scale
        val h = height * scale

        drawBackground(x, y, w, h, style, isPressed, percentFaded, scale)

        val color = style.textColor(isPressed, percentFaded)
        val shadow = if (isPressed) style.isShadowActive else style.isShadowIdle
        drawSpaceGlyph(x, y, w, h, color, shadow)
    }

    override val keyName: String
        get() = "" // not used for space rendering
}
